cmd/plugins/juju-metadata/toolsmetadata.go

```diff
@@ -29,32 +29,41 @@ func newToolsMetadataCommand() cmd.Command {
 // toolsMetadataCommand is used to generate simplestreams metadata for juju agents.
 type toolsMetadataCommand struct {
 	cmd.CommandBase
-	fetch       bool
-	metadataDir string
-	stream      string
-	clean       bool
-	public      bool
+	fetch           bool
+	metadataDir     string
+	stream          string
+	clean           bool
+	public          bool
+	preventFallback bool
 }
 
 const toolsMetadataDoc = `
 generate-agents creates the simplestreams metadata for agents.
 
-This command works by scanning a directory for agent binary tarballs from which to generate
-simplestreams agent metadata. The working directory is specified using the -d argument
-(defaults to $JUJU_DATA or if not defined $XDG_DATA_HOME/juju or if that is not defined
-~/.local/share/juju). The working directory path must contain a "tools" subdirectory.
-This "tools" directory is expected to contain a named subdirectory containing agent binary
-tarballs, and is where the resulting metadata is written.
-
-The stream for which metadata is generated is specified using the --stream parameter
-(default is "released"). Metadata can be generated for any supported stream - released,
-proposed, testing, devel.
-
-Tools tarballs can are located in either a sub directory called "releases" (legacy),
-or a directory named after the stream. By default, if no --stream argument is provided,
-metadata for agent binaries in the "released" stream is generated by scanning for 
-agent binary tarballs in the "releases" directory. By specifying a stream explicitly, 
-agent binary tarballs are expected to be located in a directory named after the stream.
+This command works by scanning a directory for agent binary tarballs from which 
+to generate simplestreams agent metadata. The working directory is specified
+using the -d argument (defaults to $JUJU_DATA or if not defined
+$XDG_DATA_HOME/juju or if that is not defined ~/.local/share/juju). The working 
+directory path must contain a "tools" subdirectory. The "tools" directory is 
+expected to contain a streams release subdirectory (released, devel, proposed) 
+containing agent binary tarballs. The output of the streams metadata will be
+written next to the release subdirectory.
+
+The stream for which metadata is generated is specified using the --stream 
+parameter (default is "released"). Metadata can be generated for any supported 
+stream - released, proposed, testing, devel.
+
+Tools tarballs can are located in either a sub directory called "releases" 
```

Comment:
While you're here, can we changes Tools to Agents were possible?

Reply:
Fixed in https://github.com/juju/juju/pull/12165